database/set_mysql_session_vars_test.go

```diff
@@ -0,0 +1,63 @@
+package database
+
+import (
+"context"
+"database/sql/driver"
+"github.com/go-sql-driver/mysql"
+"github.com/jmoiron/sqlx"
+"github.com/stretchr/testify/assert"
+"github.com/stretchr/testify/require"
+"testing"
+)
+
+func TestSetMysqlSessionVars(t *testing.T) {
+	vars := map[string][]struct {
+		name   string
+		value  any
+		expect error
+	}{
+		"UnknownVariables": {
+			// MySQL single nodes do not recognise the "wsrep_sync_wait" system variable, but MariaDB does!
+			{name: "wsrep_sync_wait", value: int64(15)}, // MySQL unknown sys var | MariaDB succeeds
+			{name: "wsrep_sync_wait", value: int64(7)},  // MySQL unknown sys var | MariaDB succeeds
+			// Just some random unknown system variables :-)
+			{name: "Icinga", value: "Icinga"},     // unknown sys var
+			{name: "IcingaDB", value: "IcingaDB"}, // unknown sys var
+		},
+		"VariablesWithCorrectValue": { // Setting system variables known by MySQL/MariaDB to a valid value
+			{name: "transaction_isolation", value: "READ-UNCOMMITTED"},
+			{name: "explain_format", value: "TRADITIONAL"},
+			{name: "explain_format", value: "JSON"},
+			{name: "completion_type", value: int64(1) /** CHAIN */},
+			{name: "default_week_format", value: int64(7)},
+		},
+		"VariablesWithInvalidValues": { // System variables set to an invalid value
+			{name: "transaction_isolation", value: "REPEATABLE-WRITE", expect: &mysql.MySQLError{Number: 1231}},
+			{name: "completion_type", value: int64(-10), expect: &mysql.MySQLError{Number: 1231}}, // Min valid val 0
+			{name: "completion_type", value: int64(10), expect: &mysql.MySQLError{Number: 1231}},  // Max valid val 2
+			{name: "explain_format", value: "IcingaDB", expect: &mysql.MySQLError{Number: 1231}},
+		},
+	}
+
+	rdb := it.MysqlDatabase()
```

Comment:
We don't want to make the `icinga-go-library` module depend on `icinga-testing` as that would mean that all its dependencies would be pulled into its users, including Docker. That's one of the reasons why `tests/` in Icinga DB is a separate Go module so that these dependencies are kept separate from the main module.

Note that I also would take the opportunity to add tests for more of the database code. Like starting a database externally, passing the connection details to the tests, for example in environment variables, then filling a database config struct with these values, and use this to connect. That would then serve as the basis for more tests in this project and also avoid the aforementioned dependency.